### PR TITLE
Concourse webhook validator

### PR DESCRIPTION
**Pull request by sienkin**
Add source and build config for `concourse-webhook-validator`
Updated all project build configs to support pipeline-based versioned builds
Updated Makefiles to ensure wheel packages are also copied

### CODE REVIEW

File: concourse-webhook-validator/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# concourse-webhook-validator
+
+Parses a provided Concourse pipeline configuration and discovers resources configured with webhooks. These resources are then compared against the defined `create-webhooks` and `delete-webhooks` jobs to validate the webhook configuration.
+
+The scripts exists with a return code of `0` on a successful validation. The scripts exists with a return code of `1` if any validation errors are found.
```
Comment (Jata2k) [inline concourse-webhook-validator/README.md]: `exists` -> `exits`

Comment (sienkin) [reply concourse-webhook-validator/README.md]: Updated

File: concourse-webhook-validator/src/concourse_webhook_validator/__init__.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    pipeline_file = args.pipeline[0]
+    pipeline_file_path = os.path.join(pipeline_base_dir, pipeline_deployment, pipeline_team, pipeline_file)
+
+    webhooks_dict = {}
+    format_output(
+        f"Checking webhooks configuration: {colours.bold}{pipeline_file}{colours.end}"
+    )
+    pipeline_config = load_pipeline_config(pipeline_file_path)
+    if validate_pipeline_config(pipeline_config) == 0:
+        webhooks_dict = parse_pipeline_resources(pipeline_config, webhooks_dict)
```
Comment (Jata2k) [inline concourse-webhook-validator/src/concourse_webhook_validator/__init__.py]: I don't think we have to reassign here. I think we can just pass in directly. I.e. `parse_pipeline_resources(pipeline_config, webhooks_dict)`. As such we can drop the return on the parse_pipeline_resources function

Comment (sienkin) [reply concourse-webhook-validator/src/concourse_webhook_validator/__init__.py]: The `parse_pipeline_resources` function can modify the structure and so the reassignment is required to ensure `webhooks_dict` is updated.

I could rename the var to make it more explicit if needed.

Edited: After discussion, a commit has been pushed to remove the unnecessary reassignment.

**Review by Jata2k — APPROVED** [review]

**Pull request merged**